azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -821,9 +821,10 @@ public synchronized FlowResourceRecommendation createFlowResourceRecommendation(
         throw new ProjectManagerException("No flow resource recommendations have been inserted.");
       }
     } catch (final SQLException ex) {
-      logger.error(INSERT_FLOW_RESOURCE_RECOMMENDATION + " failed.", ex);
-      throw new ProjectManagerException("Insert flow resource recommendation " + flowId + " for existing "
-          + "project failed. ", ex);
+      // Possibly failed due to duplicate key. If not, fetchFlowResourceRecommendation will
+      // return another exception back.
+      logger.warn("Insert flow resource recommendation " + flowId + " for existing "
```

Comment:
Thank you for adding flow ID. Is there anything else you could add here to help the person who would need to debug this in the future?

Reply:
Makes sense. Thanks for bringing this up!